server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2259,7 +2254,6 @@ public void acquireReplicaOperationPermit(final long opPrimaryTerm, final long g
                                               final ActionListener<Releasable> onPermitAcquired, final String executorOnDelay,
                                               final Object debugInfo) {
         verifyNotClosed();
-        verifyReplicationTarget();
```

Comment:
out of curiosity - what made you switch this to be checked under a permit? (I agree it's good but was wondering if some test failed)

Reply:
yes, multiple tests failed. The typical situation is that we're promoting a replica to primary but are still receiving in-flight requests from the previous primary.